### PR TITLE
Changes to support gcal

### PR DESCRIPTION
**Pull request by levb**
- Webhook call handler now receives the HTTP message, encoded as `ServerlessRequest` (JSON)
- ~~BREAKING: added `remote_webhook_auth_type` to manifest, default `"none"` (was `"secret"`)~~ default is "secret"
- Added `appsctl plugin deploy`
- Added `/apps debug kv` command
- Added `.PartialCopy()` to a few `apps` structs
- Added optional AppID argument to `/apps debug bindings`
- small fixes

### CODE REVIEW

File: apps/app.go [hunk 1]
```diff
@@ -95,6 +95,10 @@ func DecodeCompatibleApp(data []byte) (app *App, err error) {
 type OAuth2App struct {
 	ClientID     string `json:"client_id,omitempty"`
 	ClientSecret string `json:"client_secret,omitempty"`
+
+	// Data allows apps to store custom data in their OAuth configuration. A
+	// frequent use case is storing app-level service account credentials.
+	Data interface{} `json:"data,omitempty"`
```
[the remark this answers is not in the document]
Comment (hanzei) [reply apps/app.go:101]: Sounds good. Let's talk about this offline.

File: server/httpin/gateway/webhook.go [hunk 2]
```diff
@@ -11,40 +10,32 @@ import (
...
+		log = log.With("path", sreq.Path)
+
+		err = g.proxy.NotifyRemoteWebhook(appID, *sreq)
+		if err != nil {
+			return err
+		}
+
+		log.Debugf("processed remote webhook")
+		return nil
+	}()
```
[the remark this answers is not in the document]
Comment (hanzei) [reply server/httpin/gateway/webhook.go]: Created https://mattermost.atlassian.net/browse/MM-39825

**Review by hanzei — CHANGES_REQUESTED** [review]
LGTM, besides the changes in https://github.com/mattermost/mattermost-plugin-apps/pull/282

**Review by hanzei — APPROVED** [review]
That should have been an `Approve`

**Comment by levb** [comment]
@hanzei I reverted the i18n changes, feel free to review, but I am passing it on to @DHaussermann

**Comment by levb** [comment]
@larkox Can you please make sure I did the right thing with the ES translations? (I ran the "extract" make target)

**Review by DHaussermann — APPROVED** [review]
@levb seems reasonable to merge this now and test as part of a future pass.

**Pull request merged**